semgrep/semgrep/pattern_lints.py

```diff
@@ -0,0 +1,112 @@
+import itertools
+import json
+from enum import Enum
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import NamedTuple
+
+from glom import glom  # type: ignore
+from glom import PathAccessError
+
+from semgrep.dump_ast import parsed_ast
+from semgrep.error import SemgrepLangError
+from semgrep.semgrep_types import BooleanRuleExpression
+from semgrep.semgrep_types import DUMMY_SPAN
+from semgrep.semgrep_types import OPERATORS
+
+
+class ParsedPattern(NamedTuple):
+    raw: BooleanRuleExpression
+    parsed: Dict[str, Any]
+
+
+def pattern_to_json(pattern: str, lang: str) -> Dict[str, Any]:
+    # thanks for nothing, mypy
+    obj: Dict[str, Any] = json.loads(
+        parsed_ast(to_json=True, language=lang, pattern=pattern, targets_str=[])
+    )
+    return obj
+
+
+def check_equivalent_patterns(
+    pattern_either: BooleanRuleExpression, lang: str = "python"
```

Comment:
let's not default to python

Reply:
whoops :+1: good catch, thought I deleted all of those.